infrastructure/src/test/java/org/corfudb/infrastructure/management/NodeRankTest.java

```diff
@@ -29,4 +36,46 @@ public void testSortingByName(){
         assertEquals(ranks.first(), rank1);
     }
 
+    @Test
+    public void testMinQuotaRank() {
+        NodeRankByResourceQuota minQuota = NodeRankByResourceQuota.MIN_QUOTA_RANK;
+        NodeRankByResourceQuota minQuota2 = NodeRankByResourceQuota.MIN_QUOTA_RANK;
+
+        assertEquals(0, minQuota.compareTo(minQuota2));
+    }
+
+    @Test
+    public void testMinQuotaRankVsNotExceededQuota() {
+        NodeRankByResourceQuota quota = new NodeRankByResourceQuota(
+                "a",
+                new ResourceQuotaStats(ONE_HUNDRED, TEN)
+        );
+
+        SortedSet<NodeRankByResourceQuota> set = new TreeSet<>();
+        set.add(quota);
+        set.add(NodeRankByResourceQuota.MIN_QUOTA_RANK);
+
+        assertEquals(quota, set.first());
+        assertEquals(NodeRankByResourceQuota.MIN_QUOTA_RANK, set.last());
+    }
+
+    @Test
+    public void testQuotaRankOrdering(){
```

Comment:
Can we add a test where both NodeRanks have the same amount of available space and verify that the names are returned in ascending order?

Reply:
that test exists in the 3-rd pr